athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/NeptuneRecordHandler.java

```diff
@@ -158,16 +157,20 @@ protected void readWithConstraint(final BlockSpiller spiller, final ReadRecordsR
                         final Map<Object, Object> obj = graphTraversalFinal.next();
                         return (rowWriter.writeRow(block, rowNum, (Object) obj) ? 1 : 0);
                     });
-                } catch (final Exception e) {
+                } 
+                catch (final Exception e) {
                     logger.info("readWithContraint: Exception occured " + e);
                 }
             }
             logger.info("readWithConstraint: numRows[{}] numResultRows[{}]", numRows, numResultRows.get());
-        } catch (final Exception e) {
+        } 
+        catch (final Exception e) {
             logger.info("readWithContraint: Exception occured " + e);
```

Comment:
You need to re-throw this exception otherwise a customer using Athena will not see a useful error message. They might only see empty results and not even know there was an error.

Reply:
done

throwing user friendly exception:

throw new Exception("Error occurred while fetching records, please refer to cloudwatch logs for more details");

Reply:
Should throw Runtime, not Exception.